src/main/java/org/skriptlang/skript/bukkit/spark/SparkUtils.java

```diff
@@ -0,0 +1,162 @@
+package org.skriptlang.skript.bukkit.spark;
+
+import me.lucko.spark.api.Spark;
+import me.lucko.spark.api.SparkProvider;
+import me.lucko.spark.api.statistic.StatisticWindow.CpuUsage;
+import me.lucko.spark.api.statistic.StatisticWindow.MillisPerTick;
+import me.lucko.spark.api.statistic.StatisticWindow.TicksPerSecond;
+import me.lucko.spark.api.statistic.misc.DoubleAverageInfo;
+import me.lucko.spark.api.statistic.types.DoubleStatistic;
+import me.lucko.spark.api.statistic.types.GenericStatistic;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.UnknownNullability;
+
+import java.util.Arrays;
+
+/**
+ * Utility class for accessing Spark's API.
+ */
+public class SparkUtils {
+
+	private static volatile DoubleStatistic<CpuUsage> cpuProcess;
+	private static volatile DoubleStatistic<CpuUsage> cpuSystem;
+	private static volatile GenericStatistic<DoubleAverageInfo, MillisPerTick> mspt;
+	private static volatile DoubleStatistic<TicksPerSecond> tps;
+
+	@Contract(pure = true)
+	private static @NotNull Spark spark() {
+		return SparkProvider.get();
+	}
+
+	/**
+	 * Checks for MSPT support. This should be checked before using the MSPT methods.
+	 * @return whether MSPT statistics are supported.
+	 */
+	public static boolean supportsMspt() {
+		return mspt() != null;
+	}
+
+	private static @UnknownNullability GenericStatistic<DoubleAverageInfo, MillisPerTick> mspt() {
+		if (mspt == null) {
+			synchronized (SparkUtils.class) {
+				if (mspt == null)
+					mspt = spark().mspt();
+			}
+		}
+		return mspt;
+	}
+
+	/**
+	 * Get the MSPT stat info block for the given window.
+	 * {@link #supportsMspt()} should be checked prior to using this.
+	 *
+	 * @param window the window for which to gather statistics from.
+	 * @return the MSPT info block.
+	 */
+	public static @Nullable DoubleAverageInfo mspt(MillisPerTick window) {
+		GenericStatistic<DoubleAverageInfo, MillisPerTick> mspt = mspt();
+		return mspt.poll(window);
```

Comment:
You should rename either this function or the nullary overload. The idea is that it should be easy to tell that one is a getter and one performs polling.

Reply:
It cannot be renamed.

Reply:
How so?

Reply:
It is the method from Spark itself. I tried renaming it but at that rate I would be recoding the entire method which seems is completely unncessary since it works fine

Reply:
This isn't a Spark method, it's right there!

Reply:
For clarification, I mean the `mspt()` method here, not `poll()`. Poll makes complete sense.
What I mean by this one is to rename a lot of these methods like you did with the `pollCpuUsage` one. This way it's clearer what use cases are actually doing.

Reply:
Ohhh alright, will do now!